lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -165,37 +173,34 @@ def prepare_data(d, init):
 
     def apply_callback(data):
         """A callback to apply changes."""
-        orig_xlim = axes.get_xlim()
-        orig_ylim = axes.get_ylim()
+        orig_limits = {
+            name: getattr(axes, f"get_{name}lim")()
+            for name in axis_map.keys()
+        }
 
         general = data.pop(0)
         curves = data.pop(0) if has_curve else []
         mappables = data.pop(0) if has_sm else []
         if data:
             raise ValueError("Unexpected field")
 
-        # Set / General
-        (title, xmin, xmax, xlabel, xscale, ymin, ymax, ylabel, yscale,
-         generate_legend) = general
+        title = general.pop(0)
+        axes.set_title(title)
+        generate_legend = general.pop()
 
-        if axes.get_xscale() != xscale:
-            axes.set_xscale(xscale)
-        if axes.get_yscale() != yscale:
-            axes.set_yscale(yscale)
+        for i, (name, axis) in enumerate(axis_map.items()):
+            axis_min = general[4*i]
+            axis_max = general[4*i + 1]
+            axis_label = general[4*i + 2]
+            axis_scale = general[4*i + 3]
+            if getattr(axes, f"get_{name}scale")() != axis_scale:
+                getattr(axes, f"set_{name}scale")(axis_scale)
 
-        axes.set_title(title)
-        axes.set_xlim(xmin, xmax)
-        axes.set_xlabel(xlabel)
-        axes.set_ylim(ymin, ymax)
-        axes.set_ylabel(ylabel)
-
-        # Restore the unit data
-        axes.xaxis.converter = xconverter
-        axes.yaxis.converter = yconverter
-        axes.xaxis.set_units(xunits)
-        axes.yaxis.set_units(yunits)
-        axes.xaxis._update_axisinfo()
-        axes.yaxis._update_axisinfo()
+            getattr(axes, f"set_{name}lim")(axis_min, axis_max)
+            axis.set_label_text(axis_label)
+            setattr(axis, 'converter', axis.converter)
```

Comment:
That doesn't look right; it's setting the `converter` to itself, isn't it? It should be set to the converter saved at line 41 (in the original code).

Reply:
Thanks for pointing that out. That's a silly mistake. I'll change that right away

Reply:
I have a question about this. The converter is not changed anywhere, so why are we setting the converter again in the callback? In the original code, we have 

https://github.com/matplotlib/matplotlib/blob/af39f1edffcd828f05cfdd04f2e59506bb4a27bc/lib/matplotlib/backends/qt_editor/figureoptions.py#L41-46

which is then set to the same object in 
https://github.com/matplotlib/matplotlib/blob/af39f1edffcd828f05cfdd04f2e59506bb4a27bc/lib/matplotlib/backends/qt_editor/figureoptions.py#L193-194

Am I missing something?

Reply:
Good observation! I cannot tell for sure after having a quick look at the Axis code (but hopefully someone else can), but I'd guess one of two reasons:
1. It was planned to support changing units/converter, but that didn't happen (so we can safely remove those)
2. Something with setting the limits and/or scale also affects units/converter so they must be restored 

Assuming that the current code is "correct", it would be the limits (otherwise one can move the restoring to where the scale is set. But hard to say.

Reply:
Thanks, that helps. I'll wait for someone else to weigh in on this then since most other things are resolved.

Reply:
Okay. But how should I do it on my end then? Should I keep it the way it is right now? Or should I store all the axis converters in a dictionary beforehand and then set them again, something like:

```python
axis_converters = {name: axis.converter for name, axis in axis_map.items()
```

and then in the callback:
```python
axis.converter = axis_converters[name]
```

Reply:
Yes, store converters (just as you have done with units, which cannot be changed either).

Reply:
Got it. I'll do that. Thanks!